app/controllers/api/v1/todos_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Api::V1::TodosController < Api::ApiController
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found_response
+
+  def destroy
+    todo = Todo.find(params[:id])
+    user = User.find_by_auth_token(request.headers["Authorization"])
+
+    if !user
+      render json: {message: "Unauthorized"}, status: :unauthorized
+    elsif user != todo.user
+      render json: {message: "Forbidden"}, status: :forbidden
+    else
+      todo.destroy
+      head :no_content
+    end
+  end
+
```

Comment:
I think I did it the way I did it because I wanted to have a distinction between "unauthorized" and "forbidden", and then wrote my tests that way, but it's probably the case that this distinction isn't as important as preventing security issues.